# libdom: utils
# 3 files, 54 changed lines
#
# Regenerate: tools/regen-patches.sh && tools/split-patches.py
#
diff -urN upstream/libdom/src/utils/hashtable.c browser/libdom/src/utils/hashtable.c
--- upstream/libdom/src/utils/hashtable.c
+++ browser/libdom/src/utils/hashtable.c
@@ -9,12 +9,13 @@
 
 #include <stdlib.h>
 #include <string.h>
 #include <stdbool.h>
+/* stdbool removed for CW8 -- prefix provides bool */
 #include <assert.h>
 #ifdef TEST_RIG
 #include <stdio.h>
 #endif
-#include "utils/hashtable.h"
+#include "utils/hashtable.h"
 
 #include <libwapcaplet/libwapcaplet.h>
 
diff -urN upstream/libdom/src/utils/hashtable.h browser/libdom/src/utils/hashtable.h
--- upstream/libdom/src/utils/hashtable.h
+++ browser/libdom/src/utils/hashtable.h
@@ -9,7 +9,8 @@
 #ifndef dom_utils_hashtable_h_
 #define dom_utils_hashtable_h_
 
 #include <stdbool.h>
+/* stdbool.h removed for CW8 C89 -- prefix provides bool */
 #include <dom/functypes.h>
 
 typedef struct dom_hash_table dom_hash_table;
diff -urN upstream/libdom/src/utils/list.h browser/libdom/src/utils/list.h
--- upstream/libdom/src/utils/list.h
+++ browser/libdom/src/utils/list.h
@@ -4,9 +4,12 @@
  *                http://www.opensource.org/licenses/mit-license.php
  * Copyright 2009 Bo Yang <struggleyb.nku@gmail.com>
  *
- * This file contains the list structure used to compose lists. 
- * 
+ * This file contains the list structure used to compose lists.
+ *
  * Note: This is a implementation of a doubld-linked cyclar list.
+ *
+ * MacSurf: static inline bodies extracted to libdom_c89_helpers.c
+ * for CW8 C89 compatibility.
  */
 
 #ifndef dom_utils_list_h_
@@ -19,43 +22,8 @@
 	struct list_entry *next;
 };
 
-/**
- * Initialise a list_entry structure
- *
- * \param ent  The entry to initialise
- */
-static inline void list_init(struct list_entry *ent)
-{
-	ent->prev = ent;
-	ent->next = ent;
-}
-
-/**
- * Append a new list_entry after the list
- *
- * \param head  The list header
- * \param new   The new entry
- */
-static inline void list_append(struct list_entry *head, struct list_entry *new)
-{
-	new->next = head;
-	new->prev = head->prev;
-	head->prev->next = new;
-	head->prev = new;
-}
-
-/**
- * Delete a list_entry from the list
- *
- * \param entry  The entry need to be deleted from the list
- */
-static inline void list_del(struct list_entry *ent)
-{
-	ent->prev->next = ent->next;
-	ent->next->prev = ent->prev;
-
-	ent->prev = ent;
-	ent->next = ent;
-}
+extern void list_init(struct list_entry *ent);
+extern void list_append(struct list_entry *head, struct list_entry *new_entry);
+extern void list_del(struct list_entry *ent);
 
 #endif
